script/prebuild

```diff
@@ -8,7 +15,30 @@ if commits.size == 0
 elsif commits.size == 1
   puts "OK: branch contains one commit on top of #{base}"
   commit_files = `git diff-tree --no-commit-id --name-only -r #{commits[0]}`.strip.split("\n")
-  unless commit_files.include?('CHANGELOG.md')
+  uncommitted_files = `git status -s`.strip
+  unless uncommitted_files.length == 0
+    STDERR.puts "Error: there are uncommitted changes:\n #{uncommitted_files}"
+    exit 1
+  end
+  if commit_files.include?('CHANGELOG.md')
+    puts "OK: branch contains CHANGELOG.md"
+    version_file = 'src/versioning/Versioning.ts'
+    changelog_file = 'CHANGELOG.md'
+    if !!File.read(changelog_file).match?(/\[Unreleased\]/)
+      verbose("npm version patch --no-git-tag-version")
+      package = JSON.parse(File.read('package.json'))
+      version_string = package['version']
+      puts "Increasing patch version to #{version_string}"
+      File.write(version_file, File.read(version_file).gsub(/return '[.0-9]+';/, "return '#{version_string}';"))
+      File.write(changelog_file, File.read(changelog_file).gsub(/\[Unreleased\]/, "[#{version_string}] - #{Date.today}"))
```

Comment:
What will happen on the next commit? There won't be `[Unreleased]` and this block won't be run next time?

Reply:
I think my idea is for developer to add [Unreleased] and comment to CHANGELOG.md before their commit then run npm run build:release. For Travis, it won't see [Unrelease] so that block of code won't be run.